### PR TITLE
Added optional fallbacks to JSONValue GetType methods

### PR DESCRIPTION
**Pull request by MystMagus**
Have never contributed before so please tell me if I should do something differently/made any mistakes as well. I have some more stuff to add as well, just wanna be sure I'm doing this right first.

As for the change, since the GetType methods call IsType to determine if it can get the value of not anyway it makes sense to be able to get a particular default directly.

### CODE REVIEW

**Comment by ArnisLielturks** [comment]
I personally like these changes, let's see what others think about them.

**Comment by weitjong** [comment]
I think it will be clearer to call it `defaultValue`.

**Comment by MystMagus** [comment]
Makes sense.

**Comment by ArnisLielturks** [comment]
Looks good to me.

File: Source/Urho3D/Resource/JSONValue.h [hunk 1]
```diff
@@ -187,19 +187,19 @@ class URHO3D_API JSONValue
...
+    int GetInt(int defaultValue = 0) const { return IsNumber() ? (int)numberValue_ : defaultValue; }
     /// Return unsigned integer value.
-    unsigned GetUInt() const { return IsNumber() ? (unsigned)numberValue_ : 0; }
+    unsigned GetUInt(unsigned defaultValue = 0) const { return IsNumber() ? (unsigned)numberValue_ : defaultValue; }
     /// Return float value.
-    float GetFloat() const { return IsNumber() ? (float)numberValue_ : 0.0f; }
+    float GetFloat(float defaultValue = 0.0f) const { return IsNumber() ? (float)numberValue_ : defaultValue; }
     /// Return double value.
-    double GetDouble() const { return IsNumber() ? numberValue_ : 0.0; }
+    double GetDouble(double defaultValue = 0.0f) const { return IsNumber() ? numberValue_ : defaultValue; }
```
Comment (ArnisLielturks) [inline Source/Urho3D/Resource/JSONValue.h]: Minor improvement

Change 
```
GetDouble(double defaultValue = 0.0f)
```
to
```
GetDouble(double defaultValue = 0.0)
```

**Comment by ArnisLielturks** [comment]
@MystMagus are you able to provide the small fix? If no, will merge this and fix it in different commit.

**Comment by MystMagus** [comment]
Oh sorry, yes I'll take care of it.

**Pull request merged**